core/grants.go

```diff
@@ -195,3 +195,54 @@ func (a *API) revokeGrant(ctx context.Context, x apiGrant) error {
 
 	return nil
 }
+
+func (a *API) revokeGrantsByAccessToken(ctx context.Context, token string) error {
+	for _, p := range policies {
+		data, err := a.raftDB.Get(ctx, grantPrefix+p)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+
+		if data == nil {
```

Comment:
yes, but it's kind of nice to short-circuit the rest of this loop, with all the marshalling and unmarshalling, if we don't need it?